events/bingo/webApps/bingo.js

```diff
@@ -0,0 +1,484 @@
+                } else if (results.length === 0 && type === "searchOnly") {
+                    var responseObject = {
+                        status: "success",
+                        newUser: true
+                    };
+    
+                    response.statusCode = 200;
+                    response.setHeader('Content-Type', 'application/json');
+                    return response.end(JSON.stringify(responseObject));
+                } else {
+                    var responseObject = {
+                        status: "success",
+                        newUser: false,
+                        userCardNumbers: JSON.parse(results[0].cardNumbers),
+                        userCardColor: JSON.parse(results[0].cardColor)
+                    };
+    
+                    response.statusCode = 200;
+                    response.setHeader('Content-Type', 'application/json');
+                    return response.end(JSON.stringify(responseObject));
+                }
+            });
+        } else {
+            var responseObject = {
+                status: "error"
+            };
+
+            response.statusCode = 200;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+    } else if (type === "newRound") {
+        return startNewRound(queryParamObject.newTablePrefix, response);
+    } else if (type === "getCalledNumbers") {
+        var responseObject = {
+            "status": "success",
+            "calledNumbers": currentCalledNumbers
+        };
+
+        response.statusCode = 200;
+        response.setHeader('Content-Type', 'application/json');
+        return response.end(JSON.stringify(responseObject));
+    } else {
+        response.statusCode = 501;
+        response.setHeader('Content-Type', 'text/plain');
+        return response.end(JSON.stringify(queryParamObject) + '\n');
+    }
+}
+
+// A utility function for creating a "CASE" argument in an SQL
+// query when recording winners. 
+function createCaseString(winnersArray) {
+    var caseString = '';
+
+    for (var i = 0; i < winnersArray.length; i++) {
+        caseString += "when username='" + winnersArray[i].username +
+            "' then '" + winnersArray[i].prizeWon + "'";
+
+        if (i < (winnersArray.length - 1)) {
+            caseString += ' ';
+        }
+    }
+
+    return caseString;
+}
+
+// A utility function for creating a "IN" argument in an SQL
+// query when recording winners.
+function createQueryInString(winnersArray) {
+    var inString = '';
+
+    for (var i = 0; i < winnersArray.length; i++) {
+        inString += "'" + winnersArray[i].username + "'";
+
+        if (i < (winnersArray.length - 1)) {
+            inString += ', ';
+        }
+    }
+
+    return inString;
+}
+
+// Records the prizes that a given list of users has won.
+// This method is inefficient: we are storing a mostly-empty
+// "prizeWon" column in each of our tables.

```

Comment:
what would be a possible solution?  a separate prize table that can be joined?

Reply:
exactly